core/src/main/java/nl/nn/adapterframework/doc/DocWriterNew.java

```diff
@@ -324,10 +324,8 @@ public void init(XsdVersion versionTag) {
 	void init(String startClassName, XsdVersion version) {
 		this.startClassName = startClassName;
 		this.version = version;
-		if(log.isTraceEnabled()) {
-			log.trace(String.format("Initialized DocWriterNew with start element name [%s], version [%s] and output file [%s]",
-					startClassName, version.toString(), outputFileNames.get(version)));
-		}
+		log.trace("Initialized DocWriterNew with start element name [{}], version [{}] and output file [{}]",
+				startClassName, version.toString(), outputFileNames.get(version));
```

Comment:
Het gaat nog niet helemaal goed, nu vervang je de string.format alleen. De argumenten die je in je geformateerde string wilt zetten moet je via een lambda noteren. `() -> version.toString()` 
[Meer informatie hierover kan je hier vinden](https://www.baeldung.com/log4j-2-lazy-logging).

Reply:
Dit zijn simpele getters. Is het dan ook nodig om lambda's te doen? Ik heb uitleg gezocht over logging:

https://logging.apache.org/log4j/2.x/manual/api.html

Hier zie ik het volgende staan:

    logger.debug("Logging in user {} with birthday {}", user.getName(), user.getBirthdayCalendar());

    With the code above the logging level will only be checked once and the String construction will only occur when debug logging is enabled.

Reply:
Het zijn simpele getters, maar het zijn wel method aanroepen. Als je ze als het ware prefixed met '() ->', dan worden die methods alleen aangeroepen als het werkelijk nodig is.

Reply:
Ter verduidelijking; een argument meegeven aan een methode is goed en kan geen kwaad. Maar als het argument dat je meegeeft niet een pointer is naar een veld maar een andere methode aanroept, dan moet eerst die methode uitgevoerd worden voordat de daadwerkelijke (log.trace) methode aangeroepen wordt.